crates/server-fixture/server/src/lib.rs

```diff
@@ -125,3 +136,58 @@ async fn html(
 
     Html(include_str!("data/4kb.html"))
 }
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use axum::{
+        body::Body,
+        http::{self, Request, StatusCode},
+    };
+    use http_body_util::BodyExt;
+    use serde_json::Value;
+    use tower::ServiceExt;
+
+    fn get_app() -> Router {
+        let (sender, _) = oneshot::channel();
+        let state = AppState {
+            shutdown: Some(sender),
+        };
+        app(state)
+    }
+
+    #[tokio::test]
+    async fn hello_world() {
+        let response = get_app()
+            .oneshot(Request::builder().uri("/").body(Body::empty()).unwrap())
```

Comment:
this definitely worked for testing purely the request and response; tho if we want more of an integration test of the server, then the test should spin up the server like its `main.rs` (using `tokio::spawn` & `TcpListener`)